lib/spec/openapi/utils.js

```diff
@@ -68,18 +68,47 @@ function prepareOpenapiObject (opts) {
   return openapiObject
 }
 
-function normalizeUrl (url, servers, stripBasePath) {
+function normalizeUrl (url, serverUrls, stripBasePath) {
   if (!stripBasePath) return formatParamUrl(url)
-  servers = Array.isArray(servers) ? servers : []
-  servers.forEach(function (server) {
-    const basePath = new URL(server.url).pathname
+  serverUrls.forEach(function (serverUrl) {
+    const basePath = serverUrl.startsWith('/') ? serverUrl : new URL(serverUrl).pathname
     if (url.startsWith(basePath) && basePath !== '/') {
       url = url.replace(basePath, '')
     }
   })
   return formatParamUrl(url)
 }
 
+function resolveServerUrls (servers) {
+  const resolvedUrls = []
+
+  servers = Array.isArray(servers) ? servers : []
+  for (const server of servers) {
+    const originalUrl = server.url
+    const variables = server.variables
+
+    let url = originalUrl
+    const findVariablesRegex = /{(.*?)}/g // As for OpenAPI v3 spec url variables are named in brackets, e.g. {foo}
+    const matches = url.matchAll(findVariablesRegex)
+
+    for (const match of matches) {
```

Comment:
Cant we do and remove lines 95 and 96?

```suggestion
    for (const [variableNameInBrackets, variableName] of matches) {
```